packages/client/src/rpc/modules/eth.ts

```diff
@@ -1110,4 +1170,72 @@ export class Eth {
 
     return bigIntToHex(gasPrice)
   }
+
+  async feeHistory(params: [string | number | bigint, string, [bigint]?]) {
+    let [blockCount] = params
+    const [, lastBlockRequested, priorityFeePercentiles] = params
+
+    blockCount = BigInt(blockCount)
+
+    if (blockCount < 1 || blockCount > 1024) {
+      throw {
+        code: INVALID_PARAMS,
+        message: 'invalid block count',
+      }
+    }
+
+    const { number: lastRequestedBlockNumber } = (
+      await getBlockByOption(lastBlockRequested, this._chain)
+    ).header
+
+    const oldestBlockNumber = bigIntMax(lastRequestedBlockNumber - blockCount - 1n, BigInt(0))
+
+    const requestedBlockNumbers = Array.from(
+      { length: Number(blockCount) },
+      (_, i) => oldestBlockNumber + BigInt(i)
+    )
+
+    const requestedBlocks = await Promise.all(
+      requestedBlockNumbers.map((n) => getBlockByOption(n.toString(), this._chain))
+    )
+
+    const [baseFees, gasUsedRatios] = requestedBlocks.reduce(
+      (v, b) => {
+        const [prevBaseFees, prevGasUsedRatios] = v
+        const { baseFeePerGas, gasUsed, gasLimit } = b.header
+
+        prevBaseFees.push(baseFeePerGas ?? BigInt(0))
+        prevGasUsedRatios.push(Number(gasUsed) / Number(gasLimit))
+
+        return [prevBaseFees, prevGasUsedRatios]
+      },
+      [[], []] as [(bigint | undefined)[], number[]]
+    )
+
+    const londonHardforkBlockNumber = this._chain.blockchain.common.hardforkBlock(Hardfork.London)!
+    const nextBaseFee =
+      lastRequestedBlockNumber - londonHardforkBlockNumber >= -1n
+        ? requestedBlocks[requestedBlocks.length - 1].header.calcNextBaseFee()
+        : BigInt(0)
+    baseFees.push(nextBaseFee)
+
+    let rewards: bigint[][] = []
+
+    if (this.receiptsManager && priorityFeePercentiles) {
+      rewards = await Promise.all(
+        requestedBlocks.map((b) =>
+          calculateRewards(b, this.receiptsManager!, priorityFeePercentiles)
+        )
+      )
+    }
+
+    return {
+      baseFeePerGas: baseFees.map((f) =>
+        f !== undefined ? bigIntToHex(f) : bigIntToHex(BigInt(0))
```

Comment:
Is there a reason to not just put the string literal here and save ourselves a couple of conversion steps?
```suggestion
        f !== undefined ? bigIntToHex(f) : '0x'
```

Reply:
Yup good one. Note: output of `bigIntToHex(0n)` = `0x0`. So will put that there :)